Tasks/VsTest/task.json

```diff
@@ -1,198 +1,260 @@
-        },
+            "helpMarkDown": "Folder to search for the test assemblies. Defaults to $(System.DefaultWorkingDirectory).",
+            "groupName": "testSelection"
+        },     
         {
-            "name": "runInParallel",
+            "name": "testFiltercriteria",
+            "type": "string",
+            "label": "Test filter criteria",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Additional criteria to filter tests from Test assemblies. For example: `Priority=1|Name=MyTestMethod`",
+            "groupName": "testSelection",
+            "visibleRule": "testSelector = testAssemblies"
+        },        
+        {
+            "name": "uiTests",
             "type": "boolean",
-            "label": "Run In Parallel",
+            "label": "Test mix contains UI tests",
             "defaultValue": "false",
             "required": false,
-            "helpMarkDown": "Enable parallel execution of your tests.",
-            "groupName": "executionOptions"
+            "helpMarkDown": "*******",
+            "groupName": "testSelection"
         },
         {
             "name": "runOnlyImpactedTests",
             "type": "boolean",
-            "label": "Run Only Impacted Tests",
+            "label": "Run only impacted tests",
             "defaultValue": "False",
             "required": false,
             "helpMarkDown": "Automatically select, and run only the tests needed to validate the code change.",
-            "groupName": "tiaOptions"
+            "groupName": "testSelection",
+            "visibleRule": "testSelector = testAssemblies"
         },
         {
             "name": "runAllTestsAfterXBuilds",
             "type": "string",
-            "label": "Number Of Builds After Which To Run All Tests:",
+            "label": "Number of builds after which all tests should be run",
             "defaultValue": "50",
             "required": false,
             "helpMarkDown": "Number of builds after which to automatically run all tests. Test Impact Analysis stores the mapping between test cases and source code. It is recommended to regenerate the mapping by running all tests, on a regular basis.",
-            "groupName": "tiaOptions"
-        },
+            "groupName": "testSelection",
+            "visibleRule": "testSelector = testAssemblies && runOnlyImpactedTests = true"
+        },      
+        {  
+           "name": "testPlatform",  
+            "type": "pickList",  
+            "label": "Test platform version",  
+            "defaultValue": "14.0",  
+            "required": false,  
+            "helpMarkDown": "The version of Visual Studio test to use.",  
+            "groupName": "executionOptions",  
+            "options": {  
+                 "15.0": "Visual Studio 2017",  
+                 "14.0": "Visual Studio 2015"  
+             }  
+        },         
         {
-            "name": "vstestLocationMethod",
```

Comment:
is there a way we can make sure thta definition which have selected this option can fall back to latest option automatically. so as we dont break their builds?